bdb/file.c

```diff
@@ -7662,159 +7664,156 @@ static int bdb_process_unused_files(bdb_state_type *bdb_state, tran_type *tran,
-
-            /* try to parse a file version */
-            invers = strtoull(file_name_post_prefix, &endp, 16 /*base*/);
-
-            /* if no file_version was found after the prefix or the next thing
-             * after the file version isn't .blob or .data or .index */
-            if (endp == file_name_post_prefix ||
-                (strncmp(endp, blob_ext, strlen(blob_ext)) != 0 &&
-                 strncmp(endp, data_ext, strlen(data_ext)) != 0 &&
-                 strncmp(endp, qdb_ext, strlen(qdb_ext)) != 0 &&
-                 strncmp(endp, index_ext, strlen(index_ext)) != 0)) {
-                file_version = 0;
-            } else {
-                uint8_t *p_buf = (uint8_t *)&invers,
-                        *p_buf_end = p_buf + sizeof(invers);
-                struct bdb_file_version_num_type p_file_version_num_type;
+        if (!(strlen(ent->d_name) > tp_len &&
+              strncmp(ent->d_name, table_prefix, tp_len) == 0))
+            continue;
 
-                bdb_file_version_num_get(&p_file_version_num_type, p_buf,
-                                         p_buf_end);
+        const char *file_name_post_prefix;
+        unsigned long long invers;
+        char *endp;
 
-                file_version = p_file_version_num_type.version_num;
-            }
+        /* file version should start right after the prefix */
+        file_name_post_prefix = ent->d_name + tp_len;
 
-            /* brute force scan to find any files on disk that we aren't
-             * actually using */
-            if (file_version) {
-                int found_in_llmeta = 0;
-                int i;
+        /* try to parse a file version */
+        invers = strtoull(file_name_post_prefix, &endp, 16 /*base*/);
 
-                /* We have to check new. prefix for schemachange first.
-                 * See NOTE in bdb_is_new_sc_file()
-                 */
-                rc = bdb_is_new_sc_file(bdb_state, tran, bdb_state->name,
-                                        file_version, bdberr);
-                if (rc == 1) {
+        /* if no file_version was found after the prefix or the next thing
+         * after the file version isn't .blob or .data or .index */
+        if (endp == file_name_post_prefix ||
+            (strncmp(endp, blob_ext, sizeof(blob_ext) - 1) != 0 &&
+             strncmp(endp, data_ext, sizeof(data_ext) - 1) != 0 &&
+             strncmp(endp, qdb_ext, sizeof(qdb_ext) - 1) != 0 &&
+             strncmp(endp, index_ext, sizeof(index_ext) - 1) != 0)) {
+            file_version = 0;
+        } else {
+            uint8_t *p_buf = (uint8_t *)&invers,
+                    *p_buf_end = p_buf + sizeof(invers);
+            struct bdb_file_version_num_type p_file_version_num_type;
+
+            bdb_file_version_num_get(&p_file_version_num_type, p_buf,
+                                     p_buf_end);
+
+            file_version = p_file_version_num_type.version_num;
+        }
+
+        if (file_version)
```

Comment:
I think this is wrong. Schema change is not deleting files now. Should be `if (!file_version)`

Reply:
thanks for catching. we want to continue if !file_version.